.github/workflows/e2e-tests.yml

```
@@ -66,32 +66,40 @@ jobs:
         with:
             node-version: lts/*
 
+      - name: Cache Node.js dependencies
+        id: cache-node
+        uses: actions/cache@v4
+        with:
+            path: e2e/node_modules
+            key: ${{ runner.os }}-node-${{ hashFiles('e2e/package-lock.json') }}
+
       - name: Install dependencies in ./e2e to be able to run `playwright merge-reports`
+        if: steps.cache-node.outputs.cache-hit != 'true'
         run: make e2e-setup-ci
 
       - name: Download blob reports from GitHub Actions Artifacts
         uses: actions/download-artifact@v4
         with:
-          path: all-blob-reports
+          path: ./e2e/blob-report
           pattern: blob-report-shard-*
           merge-multiple: true
 
       - name: Verify downloaded artifacts
         run: |
-          echo "Contents of all-blob-reports after download:"
-          ls -R all-blob-reports
+          echo "Contents of ./e2e/blob-report after download:"
+          ls -R ./e2e/blob-report
 
-      - name: Merge into HTML report
-        run: npx playwright merge-reports --reporter html ./all-blob-reports
+      - name: Merge into HTML report using Makefile
+        run: make e2e-merge-reports REPORT_PATH=blob-report
 
       - name: Verify merged report
         run: |
-          echo "Contents of ./playwright-report after merge:"
-          ls -R ./playwright-report || echo "No report found in ./playwright-report"
+          echo "Contents of ./e2e/playwright-report after merge:"
+          ls -R ./e2e/playwright-report || echo "No report found in ./e2e/playwright-report"
 
       - name: Upload merged HTML report
         uses: actions/upload-artifact@v4
         with:
             name: merged-html-report
```

nit: keep prefix as e2e to make it easier to search and for consistency
```suggestion
            name: e2e-test-report
```

---

updated on separate commit